projects/baby_driver/src/gpio_set.c

```diff
@@ -0,0 +1,33 @@
+#include "gpio_set.h"
+
+#include <string.h>
+
+#include "babydriver_msg_defs.h"
+#include "dispatcher.h"
+#include "gpio.h"
+
+static StatusCode prv_gpio_set_callback(uint8_t data[8], void *context, bool *tx_result) {
+  uint8_t port = data[1], pin = data[2], state = data[3];
+  if (port >= NUM_GPIO_PORTS || pin >= GPIO_PINS_PER_PORT) {
+    return STATUS_CODE_INVALID_ARGS;
+  }
+  // treat any nonzero state value as high
+  if (state > 0) {
+    state = GPIO_STATE_HIGH;
+  }
+  GpioAddress gpio_address = { .port = port, .pin = pin };
+  GpioSettings gpio_settings = {
+    .direction = GPIO_DIR_OUT,
+    .state = state,
+    .resistor = GPIO_RES_NONE,
+    .alt_function = GPIO_ALTFN_NONE,
+  };
+
+  gpio_init_pin(&gpio_address, &gpio_settings);

```

Comment:
Surround this with a `status_ok_or_return` so that if it fails we alert the Python project.